schemas/log.yml

```diff
@@ -43,3 +43,28 @@
       short: Name of the logger.
       description: >
         The name of the logger inside an application. This is usually the name of the class which initialized the logger, or can be a custom name. 
+
+    - name: origin.file
+      level: core
+      type: keyword
+      example: Bootstrap.java
+      short: The file which originated the log event.
+      description: >
+        The name of the source file which originated the log event.
+
+    - name: origin.function
+      level: core
+      type: keyword
+      example: init 
+      short: The function which originated the log event.
+      description: >
+        The name of the function or method which originated the log event.
```

Comment:
Out of curiosity: If it's a method, would we expect it to also indicate what class it belongs to? Something like `Foo.bar`?

Reply:
No, it would just be the `bar` in this case. We could, however, add an optional `origin.class` for languages which support that concept.

Reply:
Got it, SGTM. Lets only add it if we already have a use case for it. But

Reply:
We actually do have a use case for logging. For java logging libraries, we usually get both the file name and the class name of the origin. See also https://docs.oracle.com/javase/7/docs/api/java/lang/StackTraceElement.html#getClassName().

I didn't add it at first because not all languages have the concept of a class. But when it's optional, why not?

Reply:
Yeah, I could name quite a few that have a similar concept ;-) In the case of Golang I would also use `.class` even though that is not 100% correct. @axw Hope you agree :-)

+1 on adding class directly.

Reply:
:) I'm fine with class as a name, but I would have made it a qualified name like `Foo.bar` as you described. @felixbarny what was the reason behind excluding the class name? Is it useful to search on them independently?

Reply:
I don't see a reason why the class name should be included `log.origin.function`, am I missing something?

We can definitely have an additional field for the class (`log.origin.class`) which contains the fully qualified class name like `org.elasticsearch.bootstrap.Bootstrap`. But we currently don't have that field in the APM schema where we have module instead, which would translate to a Java package.

I'd like to defer these potentially controversial discussions and get the lowest common denominator in which, I think, is `log.origin.file.name`, `log.origin.file.line` and `log.origin.function`.

Reply:
In the description it mentions, it can be a function or method. So if it is a method, would it be wrong to put here `Bootstrap.foo`? What I'm getting at, it would be nice if the shortcut to this discussion is that both options are fine. Note: I left out the package name on purpose, was not even thinking of it before @felixbarny brought up the example.

Reply:
That would seem a bit weird to me as the method name is foo, not Bootstrap.foo. https://docs.oracle.com/javase/7/docs/api/java/lang/StackTraceElement.html#getMethodName() would also just return the method name. 

What would be the benefit of concatenating the simple class name and the method name?

Reply:
@felixbarny I'm not too fussed if this is deferred. I don't like to sound like a broken record, but the use case I have in mind is once again profiling, and I was hoping we could find a common definition of fields.

I'd say the benefit of having them combined is having a simple display name, so that UIs don't need to know how to recombine the class and method name. e.g. it's "Class::Method" in C++.